### PR TITLE
[TRTLLM-11285][feat] Fuse indexer wk + weights_proj into single GEMM in TF32 for DS-V3.2

### PR DESCRIPTION
**Pull request by peihu-nv**
### Prerequisite                                                                                                                                                                                       
  - [x] Depends on https://github.com/NVIDIA/TensorRT-LLM/pull/11989       

## Summary by CodeRabbit

* **Refactor**
  * Optimized sparse attention computation through improved weight fusion mechanism for better performance
  * Simplified sparse attention initialization and forward computation flow

* **Tests**
  * Updated sparse attention tests to align with refactored weight fusion approach

## Description

- Fuse the two DS-V3.2 indexer projections (wk and weights_proj) into a single FP32 cuBLAS GEMM (TF32 tensor cores on Ampere+), saving one kernel launch
- Force wk to FP32 with no quantization to match weights_proj, enabling the fusion
- FP32 (TF32) chosen over FP16 based on accuracy analysis in TRTLLM-10283

## Test Coverage

- [x]  test_sparse_mla_forward.py
- [x]  test_short_seq_mha.py
- [x]  NVFP4 inference test (TP=4, DS-V3.2 checkpoint)

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/attention_backend/sparse/dsa.py [hunk 1]
```diff
@@ -1023,6 +1027,13 @@ def __init__(self,
         self.ln_events = [torch.cuda.Event(), torch.cuda.Event()]
         self.weight_scale_factor = self.softmax_scale * self.n_heads**-0.5
 
+    def post_load_weights(self):
+        """Fuse wk + weights_proj into single FP32 weight for cuBLAS GEMM (TF32 on Ampere+)."""
+        # wk: [head_dim, hidden_size] + weights_proj: [n_heads, hidden_size]
+        # → fused: [head_dim + n_heads, hidden_size]
+        self._fused_wk_wp_weight = torch.cat(
```
Comment (pengbowang-nv) [inline tensorrt_llm/_torch/attention_backend/sparse/dsa.py]: nit: Do we need to cleanup the original weight? (should be ~300mb for dsv3.2)

**Review by pengbowang-nv — APPROVED** [review]
LGTM. Leaving a minor question.
